src/topicFactory.js

```diff
@@ -3,91 +3,100 @@ import qs from 'qs'
+        window.onhashchange = () => {
+          if (window.location.hash) {
+            const params = qs.parse(window.location.hash.slice(1))
+            if (params[topicName]) {
+              window.onhashchange = () => {}
+              window.location.hash = ''
+              cb(null, params[topicName])
+            } else if (params.error) {
+              window.onhashchange = () => {}
+              window.location.hash = ''
+              cb(params.error)
             }
-          } catch (err) {
-            console.error(err.stack)
-            clearInterval(interval)
-            return cb(err)
           }
-          // Check for param, stop polling and callback if present
-          if (data && data[topicName]) {
-            clearInterval(interval)
-            clearTopic(url)
-            return cb(null, data[topicName])
-          }
-        })
-      }, pollingInterval)
-  }
+        }
+      }
 
+      const pollForResult = (topicName, url, cb) => {
+        let interval = setInterval(
 
-  function clearTopic (url) {
-    nets({
-      uri: url,
-      method: 'DELETE',
-      rejectUnauthorized: false
-    }, function (err) { if (err) { throw err } /* Errors withouth this cb */ })
-  }
+          () => {
+            nets({
+              uri: url,
+              method: 'GET',
+              rejectUnauthorized: false
+            },
+
+           (err, res, body) => {
+              if (err) return cb(err)
+              // TODO what is good error message
+              if (stopBool) return cb(new Error('Request Cancelled'))
+              // parse response into raw account
+              let data
 
-  function newTopic (topicName) {
-    let url
-    if (isOnMobile) {
-      url = window.location.href
-    } else {
-      url = chasquiUrl + randomString(16)
-    }
-    const topic = new Promise((resolve, reject) => {
-      const cb = (error, response) => {
-        if (error) return reject(error)
-        resolve(response)
+              try {
+                data = JSON.parse(body).message
+                if (data.error) {
+                  clearInterval(interval)
+                  return cb(data.error)
+                }
+              } catch (err) {
+                console.error(err.stack)
+                clearInterval(interval)
+                return cb(err)
+              }
+              // Check for param, stop polling and callback if present
+              if (data && data[topicName]) {
+                clearInterval(interval)
+                clearTopic(url)
+                return cb(null, data[topicName])
+              }
+            })
+          }, pollingInterval)
       }
-      if (isOnMobile) {
-        waitForHashChange(topicName, cb)
-      } else {
-        pollForResult(topicName, url, cb)
+
+      return {
+        stop: () => {
+          stopBool = true
+        },
+        url: url,
+        stopBool: stopBool,
+        listen:  () => (
```

Comment:
@zachferland I don't want the API to get complex again. The point of it returning a promise is to remove unnecessary moving parts. Adding this `listen()` add an extra step of complexity that as a side effect makes it much harder to test.

Instead you can achieve the same thing by adding the `stop()` function to the topic itself.

Since this is returned to the underlying code automatically they can just call stop on it there.